File: examples/android/MSDKSampleAppJava/app/src/main/java/msdk/java/sample/db/entity/Action.java
package msdk.java.sample.db.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.List;

import msdk.java.messages.QuestionMessage;

@Entity
public class Action {
    @PrimaryKey(autoGenerate = true)
    public int id;

    @ColumnInfo(name = "name")
    public String name;

    @ColumnInfo(name = "type")
    public String type = null;

    @ColumnInfo(name = "description")
    public String description;

    @ColumnInfo(name = "details")
    public String details;

    @ColumnInfo(name = "icon")
    public String icon;

    @ColumnInfo(name = "invite")
    public String invite;

    @ColumnInfo(name = "status")
    public String status;

    @ColumnInfo(name = "pwDid")
    public String pwDid;

    @ColumnInfo(name="threadId")
    public String threadId;

    @ColumnInfo(name = "entryId")
    public String entryId;

    @ColumnInfo(name = "selectedAnswer")
    public String selectedAnswer;

    @ColumnInfo(name = "messageAnswers")
    public List<QuestionMessage.Response> messageAnswers;
}

File: examples/android/MSDKSampleAppJava/app_kt/src/main/kotlin/msdk/java/samplekt/db/entity/Action.kt
package msdk.java.samplekt.db.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import msdk.java.messages.StructuredMessage

@Entity
data class Action(
    @PrimaryKey(autoGenerate = true)
    var id: Int = 0,

    @ColumnInfo(name = "name")
    var name: String? = null,

    @ColumnInfo(name = "type")
    var type: String? = null,

    @ColumnInfo(name = "description")
    var description: String? = null,

    @ColumnInfo(name = "details")
    var details: String? = null,

    @ColumnInfo(name = "icon")
    var icon: String? = null,

    @ColumnInfo(name = "invite")
    var invite: String? = null,

    @ColumnInfo(name = "status")
    var status: String? = null,

    @ColumnInfo(name = "pwDid")
    var pwDid: String? = null,

    @ColumnInfo(name = "claimId")
    var claimId: String? = null,

    @ColumnInfo(name = "threadId")
    var threadId: String? = null,

    @ColumnInfo(name = "entryId")
    var entryId: String? = null,

    @ColumnInfo(name = "selectedAnswer")
    var selectedAnswer: String? = null,

    @ColumnInfo(name = "messageAnswers")
    var messageAnswers: List<StructuredMessage.Response>? = null
)